lib/TPP/ConvertTppToLoops.cpp

```diff
@@ -22,49 +22,75 @@ using namespace mlir::tpp;
 
 namespace {
 
+static bool isScalarVal(Value val) { return !val.getType().isa<ShapedType>(); }
+
+static SmallVector<Value> getLocalIvs(Value in, Value out, ValueRange ivs,
+                                      Value zero) {
+  assert(in.getType().isa<ShapedType>());
+  assert(out.getType().isa<ShapedType>());
+  auto shapedIn = in.getType().dyn_cast<ShapedType>();
+  auto shapedOut = out.getType().dyn_cast<ShapedType>();
+  assert(shapedOut.getRank() >= shapedIn.getRank());
+
+  // Handle rank 1 input.
+  auto shapeIn = shapedIn.getShape();
```

Comment:
This looks like it could be covered on the loop below.

Reply:
I see this as an early exit. If we are dealing with rank 1 we have only two choices: zero or the right-most ivs.

Reply:
I get it, but it's a convoluted early exit. It made me think the cases were different and I had to go through to make sure they're not. This loop will run at most twice, an early exit here doesn't help much.

Reply:
You don't need to. If `shapeIn` has rank 1, the loop will only run once and `newIvs` will only have one element, `ivs[0]`. Returning `newIvs` should do the trick regardless of the rank.